### PR TITLE
Issue 32 add remove riders

### PR DESCRIPTION
**Pull request by CptMalfunction**

### CODE REVIEW

**Comment by CptMalfunction** [comment]
Note that i had to do some merging in the bluetooth rider id unit as well. So maybe i broke somthing there

**Comment by CptMalfunction** [comment]
I have not yet included any special handling for riders leaving the range of the finish id unit. The riders are added to that unit when they are recognized as on track and removed when they finish or get a DNF, so i dont think we need handling for leaving finihs range

File: CentralUnit/RaceManagement/RaceTracker.cs [hunk 1]
```diff
@@ -228,15 +262,20 @@ private void OnTimer(TimingTriggeredEventArgs args)
...
-                    EnteredEvent rider = waitingRiders.Dequeue();
+                    IdEvent rider = waitingRiders.First();
+                    waitingRiders.RemoveFirst();
+
                     TimingEvent newEvent = new TimingEvent(args.Received, rider.Rider, args.Microseconds, args.GateId);
 
                     onTrackRiders.Enqueue((rider, newEvent));
                     raceState.Enqueue(newEvent);
 
+                    startGate.RemoveKnownRider(rider.Rider.Name);
```
Comment (boons605) [inline CentralUnit/RaceManagement/RaceTracker.cs:285]: Why remove a rider from the start gate and add the rider back to the start gate at the end?
Can't we leave the rider known to the start gate or let a 'filter class' decide if a rider is added back to the start gate to allow for more flexibility?

Comment (CptMalfunction) [reply CentralUnit/RaceManagement/RaceTracker.cs:285]: We remove it so we don't get bothered by rider events if the track gets close to the start box again.

In the future whether or the rider is added back should be decided by some kind of behviour (free riding -> add back, heat -> do not add back). But right now i think its best to assume people on the outide know they shouldn't drive back to the start box during a race

Comment (boons605) [reply CentralUnit/RaceManagement/RaceTracker.cs:285]: Ok, seems like a reasonable assumption for now. Create an issue to add this behavior later?

**Review by boons605 — APPROVED** [review]

**Pull request merged**